main.go

```diff
@@ -30,44 +29,72 @@ type ObjectDetail struct {
 	LastSeen time.Time
 }
 
-// result is a ds that holds possible errors and/or object details using a channel while fetching the detail of an object
-type result struct {
-	objectDetail ObjectDetail
-	err          error
+type database struct {
+	db *sql.DB
+}
+
+type client struct {
+	cli     *http.Client
+	workers int
+	path    string
+
+	sync.RWMutex
 }
 
 var (
-	db       *sql.DB
 	host     = getenv("PSQL_HOST", "objects")
 	port     = getenv("PSQL_PORT", "5432")
 	user     = getenv("PSQL_USER", "postgres")
 	password = getenv("PSQL_PWDcas", "12345")
 	dbname   = getenv("PSQL_DB_NAME", "objects")
 )
 
+//new list of object ids
 func newObjectList() *ObjectList {
 	return &ObjectList{
 		ObjectIDs: make([]int, 0, 200),
 	}
 }
 
-func init() {
-	var err error
-	db, err = sql.Open("postgres", fmt.Sprintf(`host=%s port=%s user=%s
+//new http client for posting to path
+func newHTTPClient() *client {
+	return &client{
+		cli: &http.Client{
+			Timeout: time.Second * 5,
+		},
+		workers: 5,
+		path:    "http://host.docker.internal:9010/objects/",
```

Comment:
Simply use `http://127.0.0.1:9010` instead of host.docker.internal.

Reply:
Did that because the service expects to be ran on a docker container. No need for that when pushing the code?